presto-spi/src/main/java/com/facebook/presto/spi/predicate/TupleDomain.java

```diff
@@ -119,6 +120,46 @@ private TupleDomain(Optional<Map<T, Domain>> domains)
                 .collect(toLinkedMap(Map.Entry::getKey, entry -> new ConstantExpression(entry.getValue().getNullableSingleValue(), entry.getValue().getType()))));
     }
 
+    /**
+     * Extract all column constraints that require a set of one (or more) values in their respective Domains. All the
+     * ranges (see {@link SortedRangeSet}) must have fixed values.
+     * Returns an empty Optional if the Domain is none.
+     */
+    public static <T> Optional<Map<T, Set<NullableValue>>> extractFixedValuesFromRanges(TupleDomain<T> tupleDomain)
```

Comment:
extractFixedValuesFromRanges  -> extractFixedValueSets for clarity; 'from ranges' doesn't differentiate this method for others because they all extract information from rages;

Reply:
Done